TeensyDmx.cpp

```diff
@@ -918,14 +1044,15 @@ void TeensyDmx::respondMessage(uint16_t nackReason)
     if (m_redePin != nullptr) {
         *m_redePin = 1;
     }
-    m_dmxBufferIndex = 0;
 
     m_uart.begin(RDM_BREAKSPEED, BREAKFORMAT);
     m_uart.write(0);
     m_uart.flush();
     m_uart.begin(DMXSPEED, DMXFORMAT);
-    for (uint16_t i = 0; i < m_rdmBuffer.length; ++i) {
-        m_uart.write(reinterpret_cast<uint8_t*>(&m_rdmBuffer)[i]);
+    for (m_dmxBufferIndex = 0;
+         m_dmxBufferIndex < m_rdmBuffer.length;
+         ++m_dmxBufferIndex) {
+        m_uart.write(reinterpret_cast<uint8_t*>(&m_rdmBuffer)[m_dmxBufferIndex]);
```

Comment:
So, there's a write that takes a length.  We could use that instead.  i.e. write(&m_rdmBuffer, m_rdmBuffer.length);

Reply:
Also, m_rdmBuffer.length doesn't have the length of the header (start code and length) so that's wrong, right?

Reply:
The standard says "The Message Length value is defined as the number of slots in the RDM Packet including the START Code and excluding the Checksum." we write the checksum out separately, so that's all good. Noted on the write with length though, that sounds good.

Reply:
Switched to the write with length throughout.